### PR TITLE
New readme, docs, infra for devs

### LINKED ISSUE #1531: Update Readme with better docs
I’ve come to the determination that our documentation sucks for developers. In particular:

* Building TShock from scratch is non-trivial, at least on macOS, maybe on Windows.
* Basic system architecture (what files do what and which things operate where) is non-existent.
* The relationship between TShock, Terraria Server API, and OTAPI is not clear.
* What TShock provides as an API to other plugins and what it expects other plugins to reimplement is non-clear.

In addition, I think it’s worth having a quick start guide for how to at least get up and running with TShock in a nutshell. This doesn’t have to be in depth, but it should at least cover starting the server, authenticating, and creating a couple accounts.

RIght now, this is on my todo list for implementation later today.

### PR DESCRIPTION
**Pull request by hakusaro**
This is the WIP branch for my initiative to revamp:

- [x] The readme
- [x] The contributing guidelines
- [x] Developer setup

Please give feedback on my copy and what needs to be improved or changed. I'm trying to make it look top notch 👌🏻

(Fixes #1531.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,119 +1,115 @@
...
+          sqlite3.dll
+      
+
+1. Start `TerrariaServer.exe` and TShock will boot. Answer the startup questions, and you should be ready to roll. In the background, TShock made some folders for you. We'll come back to those later.
+
+1. Startup Terraria. Connect to a `multiplayer` server via IP and enter `localhost` if you're doing this on your local computer. If you're doing it on another computer, you need its IP address.
+
+1. Look at the server console for the _auth code_. Type `/auth [code]` (example: `/auth 12345`), then a space, then the code you see in the console in your game chat. Instead of chatting, you'll run a command on the server. This one makes you temporary admin. All commands are prefixed with `/` or `!` (to make them silent).
+
+1. Use the in-game command `/register [password]` (example: `/register lovely-ashes`) to create an account. This gives you owner rights on your server, which you can configure more to your liking later. Your `character name` is your `account name`.
```
Comment (QuiCM) [inline README.md]: Examples should be consistent.
Either
```
/register ashes
/login shank ashes
```
or
```
/register lovely-ashes
/login shank lovely-ashes
```

File: README.md [hunk 2]
```diff
@@ -1,119 +1,115 @@
...
+
+1. Login to your newly created account with `/login [username] [password]` (example: `/login shank ashes`). You should see a login success message.
+
+1. Turn off the backdoor with `/auth-verify` and your server is setup for initial use. TShock also created several files inside a new `tshock` folder. These files include `config.json` (our big configuration file), `sscconfig.json` (the server side characters configuration file), and `tshock.sqlite`. Don't lose your `tshock.sqlite` or you'll have to re-setup TShock all over again.
```
Comment (QuiCM) [inline README.md]: >Turn off the backdoor

Consider revising to "Turn off the auth system"... Mentioning backdoors is never a good idea.

------
>[...]re-setup TShock all over again

re- implies repeating, so revise to
>re-setup TShock

or
>setup TShock all over again

File: README.md (hunk 1 above)
Comment (hakusaro) [reply README.md]: 👍

File: README.md (hunk 2 above)
Comment (hakusaro) [reply README.md]: To be fair, the 'auth system' being turned off is not what we want to convey either. That's really bad language.

Comment (hakusaro) [reply README.md]: 👍

File: README.md (hunk 1 above)
Comment (QuiCM) [reply README.md]: >Your `character name` is your `account name`
Login to your newly created account with `/login [username] [password]`

Consistency would make this clearer:
>Your `character name` is your `account name`
Login to your newly created account with `/login [account name] [password]`

or
>Your `character name` is your `username`
Login to your newly created account with `/login [username] [password]`

File: README.md (hunk 2 above)
Comment (QuiCM) [reply README.md]: "Complete first-time account setup by running ...." perhaps

File: README.md (hunk 1 above)
Comment (hakusaro) [reply README.md]: Since we're moving away from `user`, I've changed this to `account name`.

File: README.md (hunk 2 above)
Comment (hakusaro) [reply README.md]: Done. Also, see #1559.

**Comment by hakusaro** [comment]
Ready for review, @Pryaxis/tshock! ✨

**Comment by hakusaro** [comment]
This bullet:

> What TShock provides as an API to other plugins and what it expects other plugins to reimplement is non-clear.

I don't think I can do right now. Too much is shifting around with Bouncer to do an accurate summation of what TShock intends to provide long-term support wise to plugins. Now isn't the time to make promises we can't keep.

**Pull request merged**